Adamant/Models/CoreData/Chatroom+CoreDataClass.swift

```diff
@@ -14,6 +14,20 @@ import CoreData
 public class Chatroom: NSManagedObject {
     static let entityName = "Chatroom"
     
+    func markMessageAsRead(id: String) {
+        guard let trs = transactions as? Set<ChatTransaction> else { return }
+
+        var unreadCount = 0
+        trs.forEach { tr in
+            if tr.txId == id {
+                tr.isUnread = false
+            }
+            unreadCount += tr.isUnread ? 1 : 0
+        }
+
+        hasUnreadMessages = unreadCount != 0
+    }
```

Comment:
I suggest that a better solution is to set isUnread directly on the transaction object instead of running a search for the specific transaction within the whole array

Reply:
We need to know the number of unread messages in the chat at this point to update the hasUnreadMessages variable.